dandischema/models.py

```diff
@@ -25,6 +35,12 @@
 else:
     from typing import Literal
 
+# Local or test deployments of dandi-api will insert URLs into the schema that refer to the domain
+# localhost, which is not a valid TLD. To make the metadata valid in those contexts, setting this
+# environment variable will use a less restrictive pydantic field that allows localhost.
+if "DANDI_ALLOW_LOCALHOST_URLS" in os.environ:
```

Comment:
```suggestion
if "DANDISCHEMA_ALLOW_LOCALHOST_URLS" in os.environ:
```
or
```suggestion
if "DANDI_SCHEMA_ALLOW_LOCALHOST_URLS" in os.environ:
```
to not conflate with `dandi-cli` variables, or shouldn't bother?

I guess if we hit more of similar tune ups needed, the whole schema might indeed need to become "configurable".  I agree with @dchiquito that import-time tune up like this makes it rather suboptimal since e.g. we wouldn't be able to do validation against "proper" schema for the assets which aren't coming from localhost while some others would be for localhost one.

Reply:
We also use `DANDI_` variables for dandi-api, I think it's fine to have that as a global prefix for everything in the DANDI ecosystem.

Reply:
sure, but `DANDI_ALLOW_LOCALHOST_URLS` sounds like something we might end up adding at some point to another component which also uses `dandischema` and then it might be an undesired effect.  anyways -- now it is 3 brains which will remember about it and warn each other if see someone adding it somewhere ;)  so no strong desires on this one